rest/route/project.go

```diff
@@ -426,7 +428,10 @@ func (h *projectIDPatchHandler) hasCommitQueuePatchDefinition(pRef *model.APIPro
 	}
 	for _, alias := range aliases {
 		if model.FromAPIString(alias.Alias) == evergreen.CommitQueueAlias {
-			return true, nil
+			// assert this alias wasn't meant to be deleted
+			if !aliasesToDelete[model.FromAPIString(alias.Alias)] {
```

Comment:
it seems to me the `aliasesToDelete` map will not contain aliases where the alias == `CommitQueueAlias` since we return true on line 420.

Reply:
I think this comment is resolved by above--we only need to consider the existing aliases if the added aliases don't include a commit queue alias. I can add some comments to make this more clear.